src/test/java/fi/aalto/cs/apluscourses/intellij/actions/GetSubmissionsDashboardActionTest.java

```diff
@@ -0,0 +1,105 @@
+    when(mockMainViewModelProvider.getMainViewModel(project).getAuthenticationViewModel().get()
+        .getAuthentication()).thenReturn(mockAuthentication);
+
+    //  course
+    Course mockCourse = mock(Course.class);
+    when(mockMainViewModelProvider.getMainViewModel(project).getCourseViewModel().get().getModel())
+        .thenReturn(mockCourse);
+    when(mockCourse.getId()).thenReturn("1");
+
+    //  action
+    GetSubmissionsDashboardAction getSubmissionsDashboardAction = new GetSubmissionsDashboardAction(
+        mockMainViewModelProvider, Notifications.Bus::notify);
+    GetSubmissionsDashboardAction spyGetSubmissionsDashboardAction = spy(
+        getSubmissionsDashboardAction);
+
+    //  submissions
+    SubmissionsDashboard mockSubmissionsDashboard = mock(SubmissionsDashboard.class);
+    doReturn(mockSubmissionsDashboard).when(spyGetSubmissionsDashboardAction)
+        .tryGetSubmissionsDashboard(1L, mockAuthentication, project);
+
+    //  when
+    spyGetSubmissionsDashboardAction.actionPerformed(anActionEvent);
+
+    //  then
+    verify(spyGetSubmissionsDashboardAction, times(1))
+        .tryGetSubmissionsDashboard(1L, mockAuthentication, project);
+    verify(mockCourse, times(1))
+        .setSubmissionsDashboard(mockSubmissionsDashboard);
+  }
+
+  @Test
+  public void testActionPerformedWithInValidDataFails() {
+    //  given
+    Project project = getProject();
+    AnActionEvent anActionEvent = mock(AnActionEvent.class);
+    doReturn(project).when(anActionEvent).getProject();
+
+    MainViewModelProvider mockMainViewModelProvider = mock(MainViewModelProvider.class,
+        RETURNS_DEEP_STUBS);
+    //  auth
+    APlusAuthentication mockAuthentication = null;
+    when(mockMainViewModelProvider.getMainViewModel(project).getAuthenticationViewModel().get()
+        .getAuthentication()).thenReturn(mockAuthentication);
+
+    //  course
+    Course mockCourse = mock(Course.class);
+    when(mockMainViewModelProvider.getMainViewModel(project).getCourseViewModel().get().getModel())
+        .thenReturn(mockCourse);
+    when(mockCourse.getId()).thenReturn("1");
+
+    //  action
+    GetSubmissionsDashboardAction getSubmissionsDashboardAction = new GetSubmissionsDashboardAction(
+        mockMainViewModelProvider, Notifications.Bus::notify);
+    GetSubmissionsDashboardAction spyGetSubmissionsDashboardAction = spy(
+        getSubmissionsDashboardAction);
+
+    //  submissions
+    SubmissionsDashboard mockSubmissionsDashboard = mock(SubmissionsDashboard.class);
+    doReturn(mockSubmissionsDashboard).when(spyGetSubmissionsDashboardAction)
+        .tryGetSubmissionsDashboard(1L, mockAuthentication, project);
+
+    //  when
+    spyGetSubmissionsDashboardAction.actionPerformed(anActionEvent);
+
+    //  then
+    verify(spyGetSubmissionsDashboardAction, times(0))
+        .tryGetSubmissionsDashboard(1L, mockAuthentication, project);
+    verify(mockCourse, times(0))
```

Comment:
Nit: Just noting that there is `never()` alias for `times(0)`. Also, `times(1)` can be omitted.